app/common/helpers/collections.py

```diff
@@ -165,6 +169,11 @@ def submit_answer_for_question(self, question_id: UUID, form: DynamicQuestionFor
         data = _form_data_to_question_type(question, form)
         interfaces.collections.update_collection_data(self.collection, question, data)
 
+    def mark_form_as_complete(self, form: "Form", user: "User") -> None:
+        interfaces.collections.add_collection_metadata(
+            self.collection, MetadataEventKey.FORM_RUNNER_FORM_COMPLETED, user, form
+        )
```

Comment:
Should we also check (and below, test) that data is actually present for all relevant questions?

Reply:
In the next PR I've added a helper method that calls the interfaces that does an integrity check to make sure all of the questions have been answers and raises a value error otherwise (with coverage!). I'll push that up and have to chat through how its feeling at the helper vs. interface level